addons/default/visiosoft/advs-module/src/Adv/Command/appendRequestURL.php

```diff
@@ -11,37 +11,42 @@ class appendRequestURL
     protected $request;
     protected $url;
     protected $new_parameters;
+    protected $remove_parameters;
 
     /**
      * appendRequestURL constructor.
      * @param $request
      */
-    public function __construct($request, $url, $new_parameters = [])
+    public function __construct($request, $url, $new_parameters = [], $remove_parameters = [])
     {
         $this->url = $url;
         $this->request = $request;
         $this->new_parameters = $new_parameters;
+        $this->remove_parameters = $remove_parameters;
     }
 
     /**
      * @return appendRequestURL
      */
     public function handle()
     {
-        if (count($this->new_parameters) === 0 && count($this->request) === 0) {
+
+        $request = $this->removeParameters($this->request);
+
+        if (count($this->new_parameters) === 0 && count($request) === 0) {
```

Comment:
This whole section looks messy. This could be written more clean and tidy. Low readable !